sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -658,6 +659,59 @@ public String getEntityPath() {
             transactionContext);
     }
 
+    /**
+     * Gets and starts the auto lock renewal for a message with the given lock.
+     *
+     * @param lockToken Lock token of the message.
+     * @param maxLockRenewalDuration Maximum duration to keep renewing the lock token.
+     * @return A lock renewal operation for the message.
+     */
+    public LockRenewalOperation getAutoRenewMessageLock(String lockToken, Duration maxLockRenewalDuration) {
+        if (isDisposed.get()) {
+            throw logger.logThrowableAsError(new IllegalStateException(
+                String.format(INVALID_OPERATION_DISPOSED_RECEIVER, "getAutoRenewMessageLock")));
+        } else if (Objects.isNull(lockToken)) {
+            throw logger.logThrowableAsError(new NullPointerException("'lockToken' cannot be null."));
+        } else if (lockToken.isEmpty()) {
+            throw logger.logThrowableAsError(new IllegalArgumentException("'lockToken' cannot be empty."));
+        } else if (receiverOptions.isSessionReceiver()) {
+            throw logger.logThrowableAsError(new IllegalStateException(
+                String.format("Cannot renew message lock [%s] for a session receiver.", lockToken)));
+        } else if (maxLockRenewalDuration == null) {
+            throw logger.logThrowableAsError(new NullPointerException("'maxLockRenewalDuration' cannot be null."));
+        } else if (maxLockRenewalDuration.isNegative()) {
+            throw logger.logThrowableAsError(new IllegalArgumentException(
+                "'maxLockRenewalDuration' cannot be negative."));
+        }
+
+        return new LockRenewalOperation(lockToken, maxLockRenewalDuration, false, this::renewMessageLock);
+    }
+
+    /**
+     * Gets and starts the auto lock renewal for a session with the given lock.
+     *
+     * @param sessionId Id for the session to renew.
+     * @param maxLockRenewalDuration Maximum duration to keep renewing the lock token.
+     * @return A lock renewal operation for the message.
+     * @throws IllegalStateException if the receiver is a non-session receiver or the receiver is disposed.
+     */
+    public LockRenewalOperation getAutoRenewSessionLock(String sessionId, Duration maxLockRenewalDuration) {
+        if (isDisposed.get()) {
+            throw logger.logThrowableAsError(new IllegalStateException(
+                String.format(INVALID_OPERATION_DISPOSED_RECEIVER, "getAutoRenewSessionLock")));
```

Comment:
Shouldn't we check for empty or null session id and throw Exception as needed?

Reply:
It's checked in the constructor for LockRenewalOperation.